ui/components/src/Legend/ListLegend.tsx

```diff
@@ -11,40 +11,156 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { List } from '@mui/material';
+import { useTheme } from '@mui/material';
+import { VariableSizeList, ListChildComponentProps } from 'react-window';
+import { useRef, useEffect, forwardRef, useCallback } from 'react';
 import { LegendItem } from '../model';
 import { ListLegendItem } from './ListLegendItem';
 
 interface ListLegendProps {
   items: LegendItem[];
+  height: number;
+  width: number;
 }
 
+// Default height used to start while virtualizing the list before a true
+// height is determined.
+const DEFAULT_ROW_HEIGHT = 26;
```

Comment:
should we make this an optional prop?

Reply:
is this used to determine the height of the skeleton row when it's loading?

Reply:
> is this used to determine the height of the skeleton row when it's loading if there is a loading state?

The virtualized list needs to know heights to lay out the list, but we can't be sure of the height of items until we try to render them. This value is used to initially lay out the items in the virtualized list on the first render. After that happens,  we're able to look at the rendered height of items and adjust appropriately (the bits where we put refs on the row and then set the height).

> should we make this an optional prop?

I was debating if that was useful, but we don't expose `ListLegend` directly. Let me take a look at if it could be a helpful prop on `Legend` that gets passed down.

Reply:
Added an `initialRowHeight` prop on `ListLegend` and exposed that on `Legend` as `listProps` (trying to follow that pattern MUI uses a bunch about passing props to a nested part of a top-level component).